### PR TITLE
moving RB status aggregating from detector to controller

### PR DESCRIPTION
**Pull request by pigletfly**
Signed-off-by: pigletfly <wangbing.adam@gmail.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes https://github.com/karmada-io/karmada/issues/1309

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
     NONE
```

### CODE REVIEW

**Comment by Garrybest** [comment]
Related to #1359

**Review by Garrybest — COMMENTED** [review]
I tink you may forgot the `ClusterResourceBindingController`.

**Comment by pigletfly** [comment]
> I tink you may forgot the `ClusterResourceBindingController`.

ResourceBindingController is the first step, and actually ClusterResourceBinding reconciles in detector handles do nothing at all.

File: pkg/controllers/binding/binding_controller.go [hunk 1]
```diff
@@ -145,9 +150,52 @@ func (c *ResourceBindingController) syncBinding(binding *workv1alpha2.ResourceBi
...
+func (c *ResourceBindingController) updateResourceStatus(binding *workv1alpha2.ResourceBinding) error {
+	resource := binding.Spec.Resource
+	gvr, err := restmapper.GetGroupVersionResource(
+		c.RESTMapper, schema.FromAPIVersionAndKind(resource.APIVersion, resource.Kind),
+	)
+	if err != nil {
+		klog.Errorf("Failed to get GVR from GVK %s %s. Error: %v", resource.APIVersion, resource.Kind, err)
+		return err
+	}
+	obj, err := c.DynamicClient.Resource(gvr).Namespace(resource.Namespace).Get(context.TODO(), resource.Name, metav1.GetOptions{})
```
Comment (RainbowMango) [inline pkg/controllers/binding/binding_controller.go:172]: Why not get the resource template from cache? 
like `helper.FetchWorkload(c.DynamicClient, c.InformerManager, c.RESTMapper, binding.Spec.Resource)`.

**Comment by RainbowMango** [comment]
/retitle moving RB status aggregating from detector to controller
/remove-kind bug
/kind cleanup

File: pkg/controllers/binding/binding_controller.go (hunk 1 above)
Comment (pigletfly) [reply pkg/controllers/binding/binding_controller.go:172]: this is the origin implement,I didn't change anything.

Comment (RainbowMango) [reply pkg/controllers/binding/binding_controller.go]: It was just updated by (#1424, @dddddai ). So I remember it.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**